crates/rspack_plugin_mf/src/container/module_federation_runtime_plugin.rs

```diff
@@ -41,20 +42,46 @@ impl RuntimeModule for FederationRuntimeModule {
     RuntimeModuleStage::Normal
   }
 
-  fn generate(&self, _: &Compilation) -> rspack_error::Result<BoxSource> {
-    Ok(RawSource::from(federation_runtime_template()).boxed())
+  fn generate(&self, compilation: &Compilation) -> rspack_error::Result<BoxSource> {
+    let chunk = compilation
+      .chunk_by_ukey
+      .expect_get(&self.chunk.expect("The chunk should be attached."));
+    Ok(RawSource::from(federation_runtime_template(chunk, compilation)).boxed())
   }
 }
 
-fn federation_runtime_template() -> String {
+fn federation_runtime_template(chunk: &Chunk, compilation: &Compilation) -> String {
   let federation_global = format!("{}.federation", RuntimeGlobals::REQUIRE);
+
+  let condition_map =
+    compilation
+      .chunk_graph
+      .get_chunk_condition_map(&chunk.ukey, compilation, chunk_has_js);
+  let has_js_matcher = compile_boolean_matcher(&condition_map);
+
+  let chunk_matcher = if matches!(has_js_matcher, BooleanMatcher::Condition(true)) {
```

Comment:
Allows me to implement custom chunk handlers in JS runtime. This was the final part - currently i cannot tell if a chunk is actual js or an internal/virtual chunk. So my chunk handler has to pass a empty chunk object callback if it doesnt know what it is, this is error prone since it may be a real error that i swallow. 

this passes the chunk match logic to a function i can access off webpack require to validate chunk ids are js chunks.